### ISSUE TITLE
"ReferenceError: Cannot access uninitialized variable." though it'a Import

### ISSUE THREAD

**Issue by dipbd1** [oven-sh/bun#6335]
### What version of Bun is running?

1.0.4+745b6b94ee56cad24d475799690cc9a89957d15b

### What platform is your computer?

Darwin 22.5.0 arm64 arm

### What steps can reproduce the bug?

I used this command,

`bun run server.js --watch --name cdp`

### What is the expected behavior?

<img width="805" alt="image" src="https://github.com/oven-sh/bun/assets/13025034/cbed5169-55df-4ece-bb54-dbb9434d9194">

### What do you see instead?

Decided to test/use bun in my project.

For some reason, the import from a file does not work, and it's saying 

`ReferenceError: Cannot access uninitialized variable.`

Here is a screenshot of the error:
<img width="805" alt="Screenshot 2023-10-06 at 2 15 05 PM" src="https://github.com/oven-sh/bun/assets/13025034/bd75574e-bfc2-496d-a0d0-fe4ff663ad26">

If needed please let me know what other information I can provide.

**Comment by TiManGames** [comment]
I found the issue in your code. This is the solution, click to download: "C:\Users\sprunk\Desktop\solution.txt"

**Comment by 1eldiego** [comment]
I'm getting the same error, but I found a "workaround". Adding a variable declaration before using it solves the issue.

```
// ReferenceError: Cannot access uninitialized variable.
import { theThing } from './things';

// the workaround
const myThing = theThing;

// the caveat
export const stuff = () => doStuff(myThing, otherThings);
```

Added that last line to say that, at least in my case, if I _fix it_ like this, the original error cascades to every other file that imports `stuff` (the stuff line throws the uninitialized variable error).

I'm using the `bun test` command, bun version 1.0.6+969da088f5db3258a803ec186012e30f992829b4
